pkg/trace/agent/agent_test.go

```diff
@@ -319,10 +318,13 @@ func TestProcess(t *testing.T) {
 			Duration: (500 * time.Millisecond).Nanoseconds(),
 			Metrics:  map[string]float64{sampler.KeySamplingPriority: 2},
 		}}
-		var traces pb.Traces
-		for size := 0; size < writer.MaxPayloadSize*payloadN; size += trace.Msgsize() {
-			traces = append(traces, trace)
-		}
+		// we are sending 3 traces
+		traces := pb.Traces{trace, trace, trace}
+		// setting writer.MaxPayloadSize to the size of 1 trace (+1 byte)
+		defer func(oldSize int) { writer.MaxPayloadSize = oldSize }(writer.MaxPayloadSize)
+		writer.MaxPayloadSize = trace.Msgsize() + 1
+		// and expecting it to result in ~2.8 payloads (3)
```

Comment:
Nit: I think it would be clearer to just write 3 rather than 2.8, I was a bit confused at what it meant.

Reply:
Haha, ok.